src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -146,19 +146,33 @@ public ChatResponse execute(String userEmail, String message, String threadId) t
       case SHEET_READ_ROW:
         int row = ((SheetReadRowAction) action).getRow();
         List<String> rowData = cloudSheet.readRow(row);
-        chatResponse = ChatResponse.createChatResponseWithList(rowData, actionType, documentUrl);
+        if (((SheetReadRowAction) action).getSelectRandomEntry()) {
+          chatResponse = new ChatResponse(selectRandomFromList(rowData), actionType, documentUrl);
+        } else {
+          chatResponse = ChatResponse.createChatResponseWithList(rowData, actionType, documentUrl);
+        }
         break;
 
       case SHEET_READ_COLUMN:
         String column = ((SheetReadColumnAction) action).getColumn();
         List<String> columnData = cloudSheet.readColumn(column);
-        chatResponse = ChatResponse.createChatResponseWithList(columnData, actionType, documentUrl);
+
+        if (((SheetReadColumnAction) action).getSelectRandomEntry()) {
+          chatResponse = new ChatResponse(selectRandomFromList(columnData), actionType, documentUrl);
+        } else {
+          chatResponse = ChatResponse.createChatResponseWithList(columnData, actionType, documentUrl);
+        }
         break;
 
       case SHEET_READ_SHEET:
         String sheetName = ((SheetReadSheetAction) action).getSheetName();
         List<List<String>> sheetData = cloudSheet.readSheet(sheetName);
-        chatResponse = ChatResponse.createChatResponseWithListList(sheetData, actionType, documentUrl);
+
+        if (((SheetReadSheetAction) action).getSelectRandomEntry()) {
+          chatResponse = new ChatResponse(selectRandomFromListList(sheetData), actionType, documentUrl);
```

Comment:
uhhh... I can't think of a use case for this... looks complex to implement as well. How about we skip it for now?

Reply:
removed!